client/ayon_core/addon/base.py

```diff
@@ -80,36 +81,41 @@ class ProcessPreparationError(Exception):
 
 
 class ProcessContext:
-    """Context of child process.
+    """Hold context of process that is going to be started.
 
-    Notes:
-        This class is used to pass context to child process. It can be used
-            to use different behavior of addon based on information in
-            the context.
-        The context can be enhanced in future versions.
+    Right now the context is simple, having information about addon that wants
+        to trigger preparation and possibly project name for which it should
+        happen.
+
+    Preparation for process can be required for ayon-core or any other addon.
+        It can be, change of environment variables, or request login to
+        a project management.
+
+    At the moment of creation is 'ProcessContext' only data holder, but that
+        might change in future if there will be need.
 
     Args:
-        addon_name (Optional[str]): Addon name which triggered process.
-        addon_version (Optional[str]): Addon version which triggered process.
+        addon_name (str): Addon name which triggered process.
+        addon_version (str): Addon version which triggered process.
         project_name (Optional[str]): Project name. Can be filled in case
             process is triggered for specific project. Some addons can have
-            different behavior based on project.
-        headless (Optional[bool]): Is process running in headless mode.
+            different behavior based on project. Value is NOT autofilled.
+        headless (Optional[bool]): Is process running in headless mode. Value
+            is filled with value based on state set in AYON launcher.
 
     """
     def __init__(
         self,
-        addon_name: Optional[str] = None,
-        addon_version: Optional[str] = None,
+        addon_name: str = None,
+        addon_version: str = None,
```

Comment:
These should now not be optional according to the type hints?
```suggestion
        addon_name: str,
        addon_version: str,
```